tests/test_version_scheme_semver2.py

```diff
@@ -1,131 +1,539 @@
-    (("1.0.0-rc.0", None, "rc", 0, None), "1.0.0-rc.1"),
-    (("1.0.0-rc.0", None, "rc", 0, 1), "1.0.0-rc.1.dev.1"),
-    (("1.0.0-rc.0", "PATCH", None, 0, None), "1.0.0"),
-    (("1.0.0-alpha.3.dev.0", None, "beta", 0, None), "1.0.0-beta.0"),
-    (("1.0.0", "PATCH", None, 0, None), "1.0.1"),
-    (("1.0.1", "PATCH", None, 0, None), "1.0.2"),
-    (("1.0.2", "MINOR", None, 0, None), "1.1.0"),
-    (("1.1.0", "MINOR", None, 0, None), "1.2.0"),
-    (("1.2.0", "PATCH", None, 0, None), "1.2.1"),
-    (("1.2.1", "MAJOR", None, 0, None), "2.0.0"),
-]
-
-local_versions = [
-    (("4.5.0+0.1.0", "PATCH", None, 0, None), "4.5.0+0.1.1"),
-    (("4.5.0+0.1.1", "MINOR", None, 0, None), "4.5.0+0.2.0"),
-    (("4.5.0+0.2.0", "MAJOR", None, 0, None), "4.5.0+1.0.0"),
-]
-
-# never bump backwards on pre-releases
-linear_prerelease_cases = [
-    (("0.1.1-beta.1", None, "alpha", 0, None), "0.1.1-beta.2"),
-    (("0.1.1-rc.0", None, "alpha", 0, None), "0.1.1-rc.1"),
-    (("0.1.1-rc.0", None, "beta", 0, None), "0.1.1-rc.1"),
-]
-
-weird_cases = [
-    (("1.1", "PATCH", None, 0, None), "1.1.1"),
-    (("1", "MINOR", None, 0, None), "1.1.0"),
-    (("1", "MAJOR", None, 0, None), "2.0.0"),
-    (("1-alpha.0", None, "alpha", 0, None), "1.0.0-alpha.1"),
-    (("1-alpha.0", None, "alpha", 1, None), "1.0.0-alpha.1"),
-    (("1", None, "beta", 0, None), "1.0.0-beta.0"),
-    (("1", None, "beta", 1, None), "1.0.0-beta.1"),
-    (("1-beta", None, "beta", 0, None), "1.0.0-beta.1"),
-    (("1.0.0-alpha.1", None, "alpha", 0, None), "1.0.0-alpha.2"),
-    (("1", None, "rc", 0, None), "1.0.0-rc.0"),
-    (("1.0.0-rc.1+e20d7b57f3eb", "PATCH", None, 0, None), "1.0.0"),
-]
-
-# test driven development
-tdd_cases = [
-    (("0.1.1", "PATCH", None, 0, None), "0.1.2"),
-    (("0.1.1", "MINOR", None, 0, None), "0.2.0"),
-    (("2.1.1", "MAJOR", None, 0, None), "3.0.0"),
-    (("0.9.0", "PATCH", "alpha", 0, None), "0.9.1-alpha.0"),
-    (("0.9.0", "MINOR", "alpha", 0, None), "0.10.0-alpha.0"),
-    (("0.9.0", "MAJOR", "alpha", 0, None), "1.0.0-alpha.0"),
-    (("0.9.0", "MAJOR", "alpha", 1, None), "1.0.0-alpha.1"),
-    (("1.0.0-alpha.2", None, "beta", 0, None), "1.0.0-beta.0"),
-    (("1.0.0-alpha.2", None, "beta", 1, None), "1.0.0-beta.1"),
-    (("1.0.0-beta.1", None, "rc", 0, None), "1.0.0-rc.0"),
-    (("1.0.0-rc.1", None, "rc", 0, None), "1.0.0-rc.2"),
-    (("1.0.0-alpha.0", None, "rc", 0, None), "1.0.0-rc.0"),
-    (("1.0.0-alpha.1", None, "alpha", 0, None), "1.0.0-alpha.2"),
-]
+from tests.utils import VersionSchemeTestArgs
 
 
 @pytest.mark.parametrize(
-    "test_input, expected",
-    itertools.chain(tdd_cases, weird_cases, simple_flow, linear_prerelease_cases),
+    "args",
+    [
+        # test driven development
+        VersionSchemeTestArgs(
+            current_version="0.1.1",
+            increment="PATCH",
+            prerelease=None,
+            prerelease_offset=0,
+            devrelease=None,
+            expected_version="0.1.2",
+        ),
```

Comment:
I feel like we should make it something like this

"version_args, expected_version"

```suggestion
        (
            VersionSchemeTestArgs(
                current_version="0.1.1",
                increment="PATCH",
                prerelease=None,
                prerelease_offset=0,
                devrelease=None,,
            ),
        "0.1.2"
```

The expected version probably should be viewed as a different argument